src/server/connection.rs

```diff
@@ -3114,6 +3142,42 @@ impl Connection {
         self.inner.send(msg.into());
         self.supported_encoding_flag = (true, not_use);
     }
+
+    #[cfg(not(any(target_os = "android", target_os = "ios")))]
+    async fn handle_window_focus(&mut self) {
+        if let Ok(displays) = super::display_service::update_get_sync_displays().await {
+            let current_display = crate::get_focused_display(displays);
```

Comment:
Instead of using the current_display in the peer info, it would be better to use a separate new message. Currently, the client handles it in `sync_peer_info` message, but the current_display doesn't actually change until the client sends a switch display command. So, it's not a true representation of peer information.